### ISSUE TITLE
Definition of Dataset

### ISSUE THREAD

**Issue by mbaudis** [phenopackets/phenopacket-schema#192]
In the definitions of [`Individual`](https://github.com/phenopackets/phenopacket-schema/blob/master/src/main/proto/base.proto#L152) and [`Biosample`](https://github.com/phenopackets/phenopacket-schema/blob/master/src/main/proto/base.proto#L228) those objects have a `datasetId` defined as "The ID of the dataset this (...object...) belongs to". Also, e.g. in Beacon the "Dataset" is an external aggregator (corresponding to a self-contained project  to which you have access etc.).

I see this as problematic, since this refers to a single local "dataset". Which doesn't make sense in the context of mix/match/re-use.

The origin of this "fixed dataset" concept is AFAIK the GA4GH schema with the VCF derived referencing of a fixed sample content model.

So, my take:

* an object can be a member of any number of datasets
* the known datasets, collections, publications ... it is a member of can be indicated through the `ExternalReferences` attribute
* I am in favour of an aggregation `Dataset` object which then references its members, not _vice versa_.

The delegation of "Dataset" to be a wrapper works well with many scenarios, e.g. the "project" concept or even a "query" concept ("all current pre-menopausal breast cancer biosamples from resource").

**Comment by pnrobinson** [comment]
Thanks for this comment. SO your suggestion is to remove the dataset id elements and to put the dataset id into the MetaData section as an External Reference? And then to allow a list of such ids?
If so, this sounds reasonable @julesjacobsen thoughts?

**Comment by mbaudis** [comment]
Complex, but basically correct. Individuals or biosamples are not "owned" by a dataset, collection, but contribute to them. They are final "bio units", not measurements, interpretations - those are derived from them. "Das Ding an sich".

The point where I see the move towards something akin to dataset is `Callset`, i.e. the output of a specific analysis using sets of normals, controls, a specific platform, algorithms ... Doesn't have to be genomic, but I am not sure what would be the corresponding aggregator in the phenotypic world? But even those can be remixed into different "Datasets" - i.e. the "BRCA" samples from "TCGA".

Long story short: Datasets sit outside. We currently handle this on the biosample level, as `external_references` (TCGA, PMID, geogsm ...). Don't ebven do it for `Individual`; there it is just implicit using the biosample annotation.

This may lead to a general 'PXF/{S}[B]/GA4GH defines "Dataset", "Collection" ... as xxx', or just to a "we'll avoid rigid dataset assignments by not doing specific attributes on the object level" resolution - but not this "biosamples are part of ->__a__<- (single) dataset".

**Comment by julesjacobsen** [comment]
Well we have the `Cohort` and `Family` concepts, which are essentially datasets of one type or another, which have their own identifiers.

I suggest we remove the `dataset_id` entirely and for that matter the `Biosample.individual_id` the latter was only RECOMMENDED and the scope of a biosample is the phenopacket. Given the phenopacket documents a single individual only, the `Biosample.individual_id` is not really necessary any more - it was when the phenopacket allowed representation of multiple individuals.
